File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/achievements/CharacterAchievementModel.ts
import type { CharacterAchievementCriteriaModel } from './CharacterAchievementCriteriaModel';
import type { AchievementModel } from './AchievementModel';

export type CharacterAchievementModel = {
    achievement: AchievementModel;
    criteria: CharacterAchievementCriteriaModel;
    completedTime: string;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Achievements/CharacterAchievementModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Achievements;

public class CharacterAchievementModel
{
    [JsonPropertyName("achievement")]
    public AchievementModel Achievement { get; set; }

    [JsonPropertyName("criteria")]
    public CharacterAchievementCriteriaModel Criteria { get; set; }

    [JsonPropertyName("completed_timestamp")]
    public long CompletedTimestamp { get; set; }
}
